src/GraphQL/Types/DateGraphType.cs

```diff
@@ -21,31 +21,37 @@ public override object Serialize(object value)
 
         public override object ParseValue(object value)
         {
-            string inputValue;
-            DateTime dateTime;
-
             if (value is DateTime)
             {
-                inputValue = ((DateTime)value).ToString("yyyy'-'MM'-'dd'T'HH':'mm':'ss.FFFFFFF'Z'");
-            }
-            else
-            {
-                inputValue = value != null ? value.ToString().Trim('"') : string.Empty;
+                DateTime dateTime = (DateTime)value;
+                return dateTime.Kind == DateTimeKind.Utc ? dateTime : dateTime.ToUniversalTime();
             }
+            
+            string inputValue = value?.ToString();
 
-            if (DateTime.TryParse(inputValue, CultureInfo.CurrentCulture, DateTimeStyles.AdjustToUniversal,
-                out dateTime))
-            {
-                return dateTime;
+            DateTime outputValue;
+            if (DateTime.TryParse(
+                inputValue,
+                CultureInfo.CurrentCulture,
+                DateTimeStyles.AdjustToUniversal,
+                out outputValue))
+            { 
+                return outputValue;
             }
+
             return null;
         }
 
         public override object ParseLiteral(IValue value)
         {
+            if (value is DateTimeValue)
+            {
+                return ((DateTimeValue)value).Value;
+            }
+
             if (value is StringValue)
             {
-                return ParseValue(((StringValue)value).Value);
+                return ParseValue(((StringValue)value).Value.Trim('"'));
```

Comment:
`Value` could be null here again.  I would move the `Trim` to the `ParseValue` method to ensure that the `Trim` happens in either case.